pinot-core/src/main/java/org/apache/pinot/server/realtime/ControllerLeaderLocator.java

```diff
@@ -229,22 +251,26 @@ private boolean isLeadControllerResourceEnabled() {
    * Thus the frequency limiting is done to guard against frequent cache refreshes, in cases where we might be getting too many NOT_SENT responses due to some other errors.
    */
   public synchronized void invalidateCachedControllerLeader() {
-    long now = System.currentTimeMillis();
+    long now = getCurrentTimeMS();
     long millisSinceLastInvalidate = now - _lastCacheInvalidateMillis;
     if (millisSinceLastInvalidate < MILLIS_BETWEEN_INVALIDATE) {
       LOGGER.info(
           "Millis since last controller cache value invalidate {} is less than allowed frequency {}. Skipping invalidate.",
           millisSinceLastInvalidate, MILLIS_BETWEEN_INVALIDATE);
     } else {
       LOGGER.info("Invalidating cached controller leader value");
-      _cachedControllerLeaderInvalid = true;
+      _cachedControllerLeaderValid = false;
       _lastCacheInvalidateMillis = now;
     }
   }
 
+  protected long getCurrentTimeMS() {
```

Comment:
Rename to `getCurrentTimeMs()`. Also, is this visible for testing?

Reply:
Updated.